examples/models/speech_recognition/TfDeepSpeech.py

```diff
@@ -0,0 +1,1137 @@
+
+        # Restore variables from training checkpoint
+        checkpoint = tf.train.get_checkpoint_state(FLAGS.checkpoint_dir)
+        checkpoint_path = checkpoint.model_checkpoint_path
+
+        output_filename = 'output_graph.pb'
+        export_temp_dir = tempfile.TemporaryDirectory()
+        export_dir = export_temp_dir.name
+
+        try:
+            output_graph_path = os.path.join(export_dir, output_filename)
+
+            def do_graph_freeze(output_file=None, output_node_names=None, variables_blacklist=None):
+                return freeze_graph.freeze_graph_with_def_protos(
+                    input_graph_def=tf.get_default_graph().as_graph_def(),
+                    input_saver_def=saver.as_saver_def(),
+                    input_checkpoint=checkpoint_path,
+                    output_node_names=output_node_names,
+                    restore_op_name=None,
+                    filename_tensor_name=None,
+                    output_graph=output_file,
+                    clear_devices=False,
+                    variable_names_blacklist=variables_blacklist,
+                    initializer_nodes='')
+
+            frozen_graph = do_graph_freeze(output_node_names=output_names, variables_blacklist='previous_state_c,previous_state_h')
+            frozen_graph.version = 1
+
+            with tf.gfile.GFile(output_graph_path, 'wb') as fout:
+                fout.write(frozen_graph.SerializeToString())
+
+            params = {}
+            # Read from temp pb file & encode it to base64 string
+            with open(output_graph_path, 'rb') as f:
+                pb_model_bytes = f.read()
+
+            params['pb_model_base64'] = base64.b64encode(pb_model_bytes).decode('utf-8')
+            return params
+
+        except RuntimeError as e:
+            logger.log('Error occured! {}'.format(e))
+
+    def load_parameters(self, params):
+        # Load the Protocol Buffers into graph def
+        tf.reset_default_graph()
+        self.graph = tf.Graph()
+        self._sess = tf.InteractiveSession(graph=self.graph)
+
+        # Load model parameters
+        pb_model_base64 = params.get('pb_model_base64', None)
+        if pb_model_base64 is None:
+            raise InvalidModelParamsException()
+        pb_model_bytes = base64.b64decode(pb_model_base64.encode('utf-8'))
+
+        graph_def = tf.GraphDef()
+        graph_def.ParseFromString(pb_model_bytes)
+        self._loaded_graph_def = graph_def
+
+
+if __name__ == '__main__':
+    test_model_class(
+        model_file_path=__file__,
+        model_class='TfDeepSpeech',
+        task=TaskType.SPEECH_RECOGNITION,
+        dependencies={
+            # Change to tensorflow-gpu to enable GPUs
+            ModelDependency.TENSORFLOW: '1.12.0',
+            # Indicate taskcluster.py file path for ds_ctcdecoder package, it will install the library based on systems
+            ModelDependency.DS_CTCDECODER: 'examples/models/speech_recognition/utils/taskcluster.py'
```

Comment:
Hmm this concept of running a script to situationally determine the exact installation of a Python module depending on the platform is quite new here. Currently, the logic of mapping a Python dependency to its exact installation (considering platform, CPU or GPU etc) is *handled by Rafiki*, not by code from the model developer. See [the related documentation](https://nginyc.github.io/rafiki/docs/latest/src/python/rafiki.client.Client.html#rafiki.client.Client.create_model), and [the related code in `rafiki/model/model.py`](https://github.com/nginyc/rafiki/blob/master/rafiki/model/model.py#L244). As such, is there a way to integrate the logic of `taskcluster.py` within Rafiki's code code instead, and have models simply indicate something like `{ ..., ModelDependency.DS_CTCDECODER: '0.6.0-alpha.1' }`, consistent with the other dependencies?

Reply:
Now can use version number. See https://github.com/nginyc/rafiki/pull/132/commits/30dd002ede0ec5c7067ecf63fffb4f92cd5943f0

Reply:
Looks great! Sorry, give me some time to test it...